pkg/registry/core/serviceaccount/storage/storage.go

```diff
@@ -52,7 +54,11 @@ func NewREST(optsGetter generic.RESTOptionsGetter, issuer token.TokenGenerator,
 
 		TableConvertor: printerstorage.TableConvertor{TableGenerator: printers.NewTableGenerator().With(printersinternal.AddHandlers)},
 	}
-	options := &generic.StoreOptions{RESTOptions: optsGetter}
+	options := &generic.StoreOptions{
+		RESTOptions: optsGetter,
+		AttrFunc: serviceaccount.GetAttrs,
+		TriggerFunc: map[string]storage.IndexerFunc{"metadata.name": serviceaccount.NameTriggerFunc},
```

Comment:
Shouldn't this be a namespaced name? This works for nodes because nodes aren't a namespaced resource, but service accounts will have name conflicts across namespaces.

Reply:
Fixed.